src/table/TablePaginationActions.jsx

```diff
@@ -46,25 +75,56 @@ export default function TablePaginationActions(props) {
     }
   };
 
+  const handleSelectPage = (event) => {
+    onPageChange(event, parseInt(event.target.value));
+  };
+
   const onFirstPage = page === 0;
   const onLastPage = page >= Math.ceil(count / rowsPerPage) - 1;
 
+  const options = [];
+  for(let i = 0; i < Math.ceil(count / rowsPerPage); i++){
+    options.push({value: i, label: i+1})
+  }
+
+  const thinTable = tableWidth < 650;
+
   return (
     <div className={classes.root}>
-      {tableWidth > 350 && (
-        <IconButton
-          onClick={!onFirstPage ? handleFirstPageButtonClick : () => {}}
-          aria-disabled={onFirstPage}
-          aria-label={translator.get('SNTable.Pagination.FirstPage')}
-          title={translator.get('SNTable.Pagination.FirstPage')}
-          tabindex={keyboardActive}
-          className={onFirstPage && classes.disabled}
+      <FormControl className={`${classes.formControl} ${thinTable && classes.hidden}`}>
+        <InputLabel className={classes.caption} htmlFor="pagination-dropdown">{translator.get('SNTable.Pagination.SelectPage')}: </InputLabel>
+        <Select
+          native
+          className={classes.dropdown}
+          id="pagination-dropdown"
+          value={page}
+          onChange={handleSelectPage}
+          label="Page"
+          inputProps={{
+            tabindex: keyboardActive,
+            name: 'Page',
+            id: 'pagination-dropdown',
+            className: classes.input,
+          }}
         >
-          <FirstPageIcon />
-        </IconButton>
-      )}
-      <IconButton
-        onClick={!onFirstPage ? handleBackButtonClick : () => {}}
+          {options.map(value => {
+            return <option value={value.value}>{value.label}</option>
+          })}
+        </Select>
+      </FormControl>
+      {tableWidth > 350 && (
+      <IconButton 
+        onClick={!onFirstPage && handleFirstPageButtonClick}
```

Comment:
<img width="779" alt="Screenshot 2021-10-14 at 12 50 10" src="https://user-images.githubusercontent.com/4471489/137303643-381ab1dd-e88a-4dc0-b660-f22c3ade727e.png">
The reason why I modified it to` !onFirstPage ? handleBackButtonClick : () => {} `is to prevent the issue above when you click the button if the button is aria-disabled.